packages/@react-stately/numberfield/src/useNumberFieldState.ts

```diff
@@ -107,17 +107,21 @@ export function useNumberFieldState(
   let prevValue = useRef(numberValue);
   let prevLocale = useRef(locale);
   let prevFormatOptions = useRef(formatOptions);
-  if (!Object.is(numberValue, prevValue.current) || locale !== prevLocale.current || formatOptions !== prevFormatOptions.current) {
-    setInputValue(format(numberValue));
-    prevValue.current = numberValue;
-    prevLocale.current = locale;
-    prevFormatOptions.current = formatOptions;
-  }
+  useEffect(() => {
```

Comment:
note, this causes a slight regression where if the numberfield is controlled and formatted, an old number can very briefly appear after commit

Reply:
FIXED